### PR TITLE
Add pg16 migration scripts (2024.7.0)

### PR DESCRIPTION
**Pull request by ggguardian**

### CODE REVIEW

File: postgresql-16-upgrade/scripts/restore.sh [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+}
+
+function usage() {
+  cat <<USAGE
+
+Usage:
+    $(basename $0) [OPTIONS]
+
+Description:
+    Backup the GitGuardian PostgresSQL database.
```
Comment (ggrosseveurgg) [inline postgresql-16-upgrade/scripts/restore.sh:86]: to update

File: postgresql-16-upgrade/scripts/migrate.sh [hunk 2]
```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env bash
+
+#####
+#
+# -- Migrate GitGuardian Database --
```
Comment (ggrosseveurgg) [inline postgresql-16-upgrade/scripts/migrate.sh:5]: do we need this script ? It seems it's included in `restore.sh` already

File: postgresql-16-upgrade/scripts/restore.sh [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+  if [[ -s "$ERROR_LOG_FILE" ]]; then
+    exit 1
+  fi
+  echo_ok
+
+  echo_step "Update backup status"
+  kubectl $KUBECTL_ARGS \
+    patch configmap postgresql-backup \
+    --type=json \
+    --patch='[{ "op": "replace", "path": "/data/status", "value": "restored" }]' >/dev/null 2>$ERROR_LOG_FILE
```
Comment (ggrosseveurgg) [inline postgresql-16-upgrade/scripts/restore.sh:246]: the CM contains a date field, it is not clear to user it matches to:
- backup (this is the case)
- restore

Should we rename that field ? Does having 2 fields (backup_date, restore_date) make sense?

Comment (ggguardian) [reply postgresql-16-upgrade/scripts/restore.sh:246]: Actually, this field could be removed as It is not read by any scripts, It is purely informational, so I suggest removing it.

File: postgresql-16-upgrade/scripts/migrate.sh (hunk 2 above)
Comment (ggguardian) [reply postgresql-16-upgrade/scripts/migrate.sh]: yes, you're right, at first, I wanted to run migrations in a separate step, so I created this script. we can definitely remove it 👍

File: postgresql-16-upgrade/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+### Update GitGuardian application
+
+From the KOTS admin console you can now update the GitGuardian application.
+
+:vertical_traffic_light: A preflight check called `PostgreSQL 16 Upgrade` will ensure that the backup was executed correctly by checking the configMap `postgresql-bakcup`.
+
+### Restore PostgreSQL backuped data
+
+Once, the applications updata has succesfully completed, you can restore the data using the `restore.sh` script:
```
Comment (rjouhann) [inline postgresql-16-upgrade/README.md:43]:
```suggestion
Once, the application update has successfully completed, you can restore the data using the `restore.sh` script:
```

**Review by rjouhann — APPROVED** [review]

File: postgresql-16-upgrade/README.md (hunk 4 above)
Comment (ggguardian) [reply postgresql-16-upgrade/README.md:43]: fixed! thanks

File: postgresql-16-upgrade/scripts/restore.sh (hunk 1 above)
Comment (ggrosseveurgg) [reply postgresql-16-upgrade/scripts/restore.sh:86]: this helper is for backup

Comment (ggguardian) [reply postgresql-16-upgrade/scripts/restore.sh:86]: Oooops! 😄

File: postgresql-16-upgrade/scripts/restore.sh [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+#####
+
+#set -x
+set -ueo pipefail
+
+# --- Variables
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+OS="$(uname -s | tr '[:upper:]' '[:lower:]')"
+DATE_ISO=$(date '+%F %T')
```
Comment (ggrosseveurgg) [inline postgresql-16-upgrade/scripts/restore.sh:20]: :face_in_clouds:

**Review by ggrosseveurgg — APPROVED** [review]

**Pull request merged**